src/middlewares/associations.js

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import db               from '../models/db';
+import * as ERR         from '../errors';
+import { associations } from '../constants';
+
+export default (req, res, next) => {
+  // Removes starting and trailing '/' and splits url in path resources.
+  const resources = req.originalUrl.replace(/^\/?|\/?$/, '').split('/');
+
+  // We only care about nested resource paths (i.e. /A(1)/B(1)), navigation
+  // links (i.e. /A(1)/B) or association links (i.e. /A(1)/B/$ref) , so we
+  // just bail out if we are dealing with a simple resource path
+  // (i.e. /A or /A(1)).
+  if (resources.length <= 1) {
+    return next();
+  }
+
+  let entities = [];
+
+  // We need to check that all associations are valid.
+  let previous;
+  while (resources.length) {
+    const resource = resources.shift();
+    if (!previous) {
+      previous = resource;
+      continue;
+    }
+
+    const previousEntity = previous.match(/(\w*)\((\d+)\)?/);
```

Comment:
This would match things like Sensors(1 or even only "(1)". I think we need something like `^(\w+)(?:\((\d+)\))?$`